functions/source/soci-index-generator-lambda/handler.go

```diff
@@ -291,38 +324,47 @@ func buildIndex(ctx context.Context, dataDir string, sociStore *store.SociStore,
 	if err != nil {
 		return nil, err
 	}
-
-	builder, err := soci.NewIndexBuilder(containerdStore, sociStore, artifactsDb, soci.WithPlatform(platform))
-	if err != nil {
-		return nil, err
+	builderOpts := []soci.BuilderOption{
+		//soci.WithMinLayerSize(0),
+		soci.WithBuildToolIdentifier("AWS SOCI CLI v0.2"),
+		soci.WithArtifactsDb(artifactsDb),
 	}
 
-	// Build the SOCI index
-	index, err := builder.Build(ctx, image)
+	builder, err := soci.NewIndexBuilder(containerdStore, sociStore, builderOpts...)
 	if err != nil {
 		return nil, err
 	}
 
-	// Write the SOCI index to the OCI store
-	err = soci.WriteSociIndex(ctx, index, sociStore, artifactsDb)
-	if err != nil {
-		return nil, err
-	}
+	// Build the SOCI index based on the specified version
+	if sociIndexVersion == "V2" {
+		// Use Convert() for V2 index generation
+		convertedOCIIndex, err := builder.Convert(ctx, image, soci.ConvertWithPlatforms(platform))
```

Comment:
Does this handle multi-arch images? I don't think we should set a platform in those cases.